### ISSUE TITLE
How to survive docker daemon restarts?

### ISSUE THREAD

**Issue by marcusholl** [SAP/devops-docker-cx-server#16]
We would like to be able to survive restarts of the docker daemon - or even machine restarts ( which implies a re-start of the docker daemon).

The Jenkins server should be up and running afterwards again.

A straight forward approach would be to provide the `--restart` flag when the Jenkins (maybe the same for Nexus) is launched. After calling `./cxserver start` the restart policy is:

```
"RestartPolicy": {
                "Name": "no",
                "MaximumRetryCount": 0
            },
```

and in fact Jenkins is not up and running after reboots.

Any other hints (e.g. implementing a call to `./cxserver start as some kind of a service (which sounds like overkill to me in the light of the `--restart` flag, but nevertheless technically possible)) welcome.

**Comment by fwilhe** [comment]
Wondering: How often do you restart a server? Sure, it is required for kernel updates (but I dimly remember RedHat and Suse working on making this obsolete?).

I'd like to learn more about the requirement before going into any implementation.

**Comment by marcusholl** [comment]
We have complains from a "real" customer. From time to time the Jenkins simply stops working. The machine on that the docker daemon is running on gets re-booted from time to time. Not fully sure about the reason, but that is how it is. Responsible for maintaining the server is an external service provider. I guess it is hard to fight against this.

Beside that specific case here: In case a server reboot happens for whatever reason I would expect that the services (in terms of the classical service notion (e.g `/etc/init.d`, as well as in the notion of applications running on that server) are up and running afterwards.

**Comment by marcusholl** [comment]
See also #18

**Comment by marcusholl** [comment]
solved with #18
